src/ResourceManager/Network/Commands.Network/NetworkWatcher/RemoveAzureNetworkWatcherCommand.cs

```diff
@@ -12,31 +12,50 @@
 // limitations under the License.
 // ----------------------------------------------------------------------------------
 
+using Microsoft.Azure.Commands.Network.Models;
 using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
 using Microsoft.Azure.Management.Network;
+using System;
 using System.Management.Automation;
 
 namespace Microsoft.Azure.Commands.Network
 {
     [Cmdlet(VerbsCommon.Remove, "AzureRmNetworkWatcher", SupportsShouldProcess = true)]
     public class RemoveAzureNetworkWatcherCommand : NetworkWatcherBaseCmdlet
     {
-        [Alias("ResourceName")]
+        [Parameter(
+             Mandatory = true,
+             ValueFromPipeline = true,
+             HelpMessage = "The network watcher resource.",
+             ParameterSetName = "SetByResource")]
```

Comment:
@junyezhu same comment as the `Location` parameter: should this be `RemoveByResource`? This applies to all of the parameters in this file

Reply:
I think Set makes sense, since we need to first set the resource, then we can add, get or remove.

Reply:
@junyezhu same comment here